### PR TITLE
1.8 Changes (Rebase 2).

### PR DESCRIPTION
**Pull request by ProgrammerDan**
Squashing because of line ending changes, so we can cleanly merge into current master.

I'm not sure if the line-change endings were altered in the XP changes or in the IO refactor (more likely IO refactor); either way:

I'm guilty of this too, but be careful with your line endings. If you are on windows, remember to issue:

    git config core.autocrlf true

or

    git config --global core.autocrlf true

if you're always working with cross platform developers.

So, this is a 100% FF only representation of the XP, new weapon and armor factories, and addresses the following issues:

* Iron XP Recipe 0 was still 32 Emeralds, fixed to 32 Bottles

* The Repair Factory configuration was not being loaded (IO Refactor issue)

* All line endings are *NIX standard now, as enforced by IO Refactor.

### CODE REVIEW

**Comment by ProgrammerDan** [comment]
@ttk2 this is a merge-ready clean apply of latest changes from ribagi's pull. It integrates fully with current master. Sadly Ribagi's branch was so far diverged from master that a fast-forward merge was impossible, and my local attempts showed that this approach was cleaner as the line endings in Ribagi's commit were all ^M (dos standard) and master were all \n (unix standard), which made git very unhappy.

I went through every file by hand that was impacted by this. It's possible I missed something. I will be testing in a moment but just in case anyone else was contemplating working on it wanted to get this pushed so they would stop and look at this instead.

File: config.yml [hunk 1]
```diff
@@ -322,14 +333,14 @@ production_factories:
...
       Baked Potato:
         material: BAKED_POTATO
-        amount: 2
+        amount: 5
       Cookie:
         material: COOKIE
-        amount: 4
+        amount: 10
       Bread:
         material: BREAD
```
Comment (MonsieurWTF) [inline config.yml]: Bread was not doubled here. (should be noted it had not specified an amount, so I assume the factory defaulted to 1 bread previously)

Comment (ProgrammerDan) [inline config.yml]: I can't comment on whether this was intentional or not, but this is perfectly consistent with the latest changes in merge request #86 -- lines [334](https://github.com/Civcraft/FactoryMod/pull/86/files#diff-0f1be8eec416e71c28b41efb0544df26R334), 338, and 341. 

That said, I concur, I'll go ahead and make bread values consistent -- good catch! Seems we all missed it so far :D .

Comment (ProgrammerDan) [inline config.yml]: Change made. I left pie alone; should that be 2 as well you figure?

**Comment by MonsieurWTF** [comment]
*shrug* That one could probably remain the same

**Comment by erocs** [comment]
Please do try to keep massive whitespace updates in their own patch so real changes in a patch can be easily seen.

**Comment by ProgrammerDan** [comment]
@erocs 

To what are you referring? Point me to lines, your contention is unclear

**Comment by erocs** [comment]
The entirety of https://github.com/ProgrammerDan/FactoryMod/commit/2d0986aa70 is one big whitespace modification. Are there also material changes which affect the configuration? I'm mentioning that it's good to keep these massive whitespace refactorings isolated to their own patch under that assumption.  :)

**Comment by ProgrammerDan** [comment]
@erocs I'll paste a screen of the diff I get from github, I see literally 0 whitespace modifications. I'm guessing you're seeing them locally? If so, pull civcraft FactoryMod master; all the whitespace modifications were IO Refactor's fault, I've simply "fixed" the XP recipes to match.

Screenshot of Diff in Github:

http://imgur.com/xK1G9pW,O2j3VqC,bTXOexD#0

Literally there are 8 whitespace modifications, if you diff against Civcraft's FactoryMod/master *as it exists now* after IO refactor.

Also be sure you have autocrlf set to true locally.

**Comment by erocs** [comment]
My mistake then, thanks!

**Comment by ProgrammerDan** [comment]
Sorry man, I feel like I'm overreacting all over the place. Were you able to confirm what I said, though? I'd hate for this to be legit a whitespace screwup :(

**Comment by erocs** [comment]
Lol, no worries it's me.

**Comment by ProgrammerDan** [comment]
Thanks man.

File: config.yml [hunk 2]
```diff
@@ -7749,3 +8059,2745 @@ production_recipes:
...
+      Potatoes:
+        material: POTATO_ITEM
+        amount: 64
+      Glass Bottle:
+        material: GLASS_BOTTLE
+        amount: 4
+    outputs:
+      Emerald:
+        material: EMERALD
+        amount: 4
```
Comment (ProgrammerDan) [inline config.yml]: TODO: Fix this, should be XP Bottles

**Comment by MonsieurWTF** [comment]
I apologize for not bringing this up sooner, but I wanted to check whether or not we should detach the repair factories from the Cauldron aspect of this update, as there are a lot of things that can be modified with how the Enchanting system works.

First of all, there is an 'Upgrades' parameter for armor/weapons/tools that are considered as inputs, but not consumed. This removes the necessity for the input-output parameters, as well as allows existing tools with durability damage to remain with their present durability.

It also allows for enchantments to be stacked without involving combination cost. at the Anvil.

**Comment by ProgrammerDan** [comment]
probably a little late at this point, sorry man

**Comment by ttk2** [comment]
we end up doing that it will have to be to later. No last minute changes.

**Comment by MonsieurWTF** [comment]
Sorry bout that. I can put that in the pipeline for future editing after all present pulls are done with.

**Comment by ProgrammerDan** [comment]
This has been replaced by #106

**Pull request closed without merging**